### ISSUE TITLE
Warning text icon clipped in Firefox v73+ when using high contrast mode or overriding colours

### ISSUE THREAD

**Issue by 36degrees** [alphagov/govuk-frontend#1863]
In the warning text icon, we use a 3px border (which is usually visually indistinct from the background) in order to better support users who use High Contrast Mode or override colours:

<img width="528" alt="Screenshot 2020-07-15 at 14 02 20" src="https://user-images.githubusercontent.com/121939/87548114-f4f6b800-c6a3-11ea-95b9-bd567fb76515.png">

https://github.com/alphagov/govuk-frontend/blob/36edb2f5eb9447a82e195bbbc0ee31c2f6ee7bff/src/govuk/components/warning-text/_index.scss#L29-L31

Firefox 73 [introduced a new 'readability backplate' feature](https://www.mozilla.org/en-US/firefox/73.0/releasenotes/):

> Many users with low vision rely on Windows' High Contrast Mode to make websites more readable. Traditionally, to increase the readability of text, Firefox has disabled background images when High Contrast Mode is enabled. With today’s release of Firefox 73, we introduce a “readability backplate” solution which places a block of background color between the text and background image. Now, websites in High Contrast Mode are more readable without disabling background images.

The new solid 'block' of background colour causes a clipping effect around the warning text icon in Firefox 73+.

<img width="528" alt="Screenshot 2020-07-15 at 14 03 14" src="https://user-images.githubusercontent.com/121939/87548147-fd4ef300-c6a3-11ea-97d4-5be069e2e1c4.png">

**Comment by 36degrees** [comment]
We could consider using an SVG for the icon.

**Comment by 36degrees** [comment]
After #2186 is merged, this should be fixed as and when / if Firefox implements the [`forced-color-adjust`](https://caniuse.com/mdn-css_properties_forced-color-adjust) property – for which [there are positive signals](https://github.com/mozilla/standards-positions/issues/463).

**Comment by 36degrees** [comment]
Firefox 89 [introduces support for the `forced-colors` CSS media](https://hacks.mozilla.org/2021/06/looking-fine-with-firefox-89/), but seemingly not `forced-color-adjust`.

Looks like https://bugzilla.mozilla.org/show_bug.cgi?id=1591210 is the bug to watch for that property being implemented.

**Comment by 36degrees** [comment]
Fixed in Firefox 113 Beta:

> The forced-color-adjust property is now supported, allowing authors to opt an element out of color changes in Forced Color Mode for improved readability where the automatically-picked contrasting colors are not ideal.

https://www.mozilla.org/en-US/firefox/113.0beta/releasenotes/

![Screenshot 2023-04-26 at 09 59 44](https://user-images.githubusercontent.com/121939/234525554-00448339-e394-4a52-b413-fcc086ed740a.png)
![Screenshot 2023-04-26 at 10 00 24](https://user-images.githubusercontent.com/121939/234525557-17bb55ae-33ec-4414-b4e5-b78c4172bd45.png)

Firefox 113 [should be released on May 9](https://wiki.mozilla.org/Release_Management/Calendar).

**Comment by 36degrees** [comment]
I'm going to close this as – other than waiting for Firefox 113 to be released – the issue has been resolved and there's nothing for us to do.
